pkg/operator/controller/canary/controller.go

```diff
@@ -110,14 +112,53 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 			if !ok {
 				return false
 			}
-			// if Spec.Port has changed, do not trigger a reconcile
-			return cmp.Equal(oldRoute.Spec.Port, newRoute.Spec.Port)
+
+			// Check if the port in newRoute is one of the ports exposed by the canary service.
+			isValidPort := false
+			for _, portNum := range CanaryPorts {
+				switch newRoute.Spec.Port.TargetPort.Type {
+				case intstr.Int:
+					if newRoute.Spec.Port.TargetPort.IntVal == portNum {
+						isValidPort = true
+						break
```

Comment:
This break doesn't do what you probably think it does—it breaks out of the `switch`, not out of the `for` loop.